xwiki-platform-core/xwiki-platform-dashboard/xwiki-platform-dashboard-ui/src/main/resources/Dashboard/XWikiUserDashboardSheet.xml

```diff
@@ -175,4 +175,146 @@
+      </async_enabled>
+      <content>
+        <disabled>0</disabled>
+        <editor>Text</editor>
+        <name>content</name>
+        <number>1</number>
+        <prettyName>Executed Content</prettyName>
+        <rows>25</rows>
+        <size>120</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </content>
+      <extensionPointId>
+        <disabled>0</disabled>
+        <name>extensionPointId</name>
+        <number>5</number>
+        <prettyName>Extension Point ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </extensionPointId>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>6</number>
+        <prettyName>Extension ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parameters>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>parameters</name>
+        <number>7</number>
+        <prettyName>Extension Parameters</prettyName>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </parameters>
+      <scope>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>scope</name>
+        <number>8</number>
+        <prettyName>Extension Scope</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>wiki=Current Wiki|user=Current User|global=Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </scope>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <content></content>
+    </property>
+    <property>
+      <extensionPointId>org.xwiki.plaftorm.user.profile.menu.content</extensionPointId>
+    </property>
+    <property>
+      <name>org.xwiki.platform.user.profile.entry.userdashboardsheet</name>
+    </property>
+    <property>
+      <parameters>id=dashboard&#xd;
+sheet=Dashboard.XWikiUserDashboardSheet&#xd;
+glyphicon=th&#xd;
+display=#if(($services.model.resolveDocument($xcontext.user) == $doc.documentReference) &amp;&amp; $hasDashboard)true#{else}false#end</parameters>
```

Comment:
Could be good to add a comment on it to explain the condition.

Reply:
I've added xml comments above each parameters section with non-trivial display conditions.

Reply:
Please don't add XML comments, we should not change the XML, it's auto-generated.

Reply:
For me a `display` parameter contains the content to display. If this is meant to be a boolean to decide if the UIX should be displayed then I's use something more explicit such as `isActive`.

Reply:
> Please don't add XML comments, we should not change the XML, it's auto-generated.

I tried to put the comment inline using `##` but I make the whole expression evaluation fails. What's the right place to put such comment?

Reply:
> What's the right place to put such comment?

Not sure there is. Could be good to check how we parse the parameters, I guess we could improve it to avoid parsing lines starting with `##`.

Reply:
Another idea is to put the logic of "display" (isActive) inside the content and don't display anything if the evaluated content is empty.

Reply:
> Another idea is to put the logic of "display" (isActive) inside the content

It's too late since you also want to not display the button.

Reply:
> I tried to put the comment inline using ## but I make the whole expression evaluation fails.

This might be something to fix.

Reply:
> Another idea is to put the logic of "display" (isActive) inside the content and don't display anything if the evaluated content is empty

I guess this is doable but it requires to pre-calculate the content of all UIX before deciding to include it or not. Also the result of `UIExtension.execute()` is a `Block` object, do a way to identify if a Block is empty exists?

So in summary I see three candidate solutions:
- @vmassol proposal 
- fixing the parsing of the parameters to ignore end of lines with `##`
- Adding a specific field in the UIX Class that would explicit this logic

Reply:
> pre-calculate the content of all UIX before deciding to include it or not.

-1 for that, we already have an issue related to execute each section on demand instead of the current "execute all" so we should definitely not rely on it